File: src/api/v1/schemas/hotspot.py
"""API Schemas - 热点分析相关"""

from __future__ import annotations

from typing import Any, Optional

from pydantic import BaseModel, Field


class HotspotTaskAccepted(BaseModel):
    task_id: str
    status: str = "pending"
    message: str = ""


class ChainStockResponse(BaseModel):
    symbol: str
    name: str
    role: str = ""
    reason: str = ""


class AnalyzedThemeResponse(BaseModel):
    title: str
    investment_logic: str = ""
    actionability: str = "medium"
    upstream: list[ChainStockResponse] = []
    midstream: list[ChainStockResponse] = []
    downstream: list[ChainStockResponse] = []
    value_flow: str = ""


class ThemeResponse(BaseModel):
    title: str
    summary: str = ""
    relevance_score: int = 0
    catalyst: str = ""
    timeline: str = ""
    risk: str = ""


class HotspotTaskInfoResponse(BaseModel):
    task_id: str
    status: str
    progress: int = 0
    message: Optional[str] = None
    current_theme: int = 0
    total_themes: int = 0
    error: Optional[str] = None
    created_at: str
    started_at: Optional[str] = None
    completed_at: Optional[str] = None


class BriefingResponse(BaseModel):
    task_id: str
    analysis_date: str = ""
    themes: list[ThemeResponse] = []
    analyzed_themes: list[AnalyzedThemeResponse] = []
    briefing: str = ""
    errors: list[str] = []


class DuplicateHotspotErrorResponse(BaseModel):
    error: str = "duplicate_hotspot"
    message: str
    existing_task_id: str


class HotspotHistoryResponse(BaseModel):
    total: int
    tasks: list[HotspotTaskInfoResponse]

File: web/src/types/hotspot.ts
export interface ChainStock {
  symbol: string;
  name: string;
  role: string;
  reason: string;
}

export interface AnalyzedTheme {
  title: string;
  investment_logic: string;
  actionability: 'high' | 'medium' | 'low';
  upstream: ChainStock[];
  midstream: ChainStock[];
  downstream: ChainStock[];
  value_flow: string;
}

export interface HotTheme {
  title: string;
  summary: string;
  relevance_score: number;
  catalyst: string;
  timeline: string;
  risk: string;
}

export interface HotspotTaskInfo {
  task_id: string;
  status: 'pending' | 'collecting' | 'extracting' | 'analyzing' | 'generating' | 'completed' | 'failed';
  progress: number;
  message: string | null;
  current_theme: number;
  total_themes: number;
  error: string | null;
  created_at: string;
  started_at: string | null;
  completed_at: string | null;
}

export interface HotspotResult {
  themes: HotTheme[];
  analyzed_themes: AnalyzedTheme[];
  briefing: string;
  errors: string[];
  analysis_date: string;
}

export type HotspotStage = 'idle' | 'collecting' | 'extracting' | 'analyzing' | 'generating' | 'completed' | 'failed';

export const HOTSPOT_STAGE_LABELS: Record<HotspotStage, string> = {
  idle: '等待开始',
  collecting: '采集市场数据',
  extracting: '提取投资主题',
  analyzing: '产业链分析',
  generating: '生成研报',
  completed: '分析完成',
  failed: '分析失败',
};
